src/ReportPortal.Client/Resources/ServiceBaseResource.cs

```diff
@@ -115,7 +115,12 @@ private void CheckSuccessStatusCode(HttpResponseMessage response, Stream stream)
                 using (var reader = new StreamReader(stream))
                 {
                     string body = reader.ReadToEnd();
-                    throw new ReportPortalException($"Response status code does not indicate success: {response.StatusCode} ({(int)response.StatusCode}) {response.RequestMessage.Method} {response.RequestMessage.RequestUri}", new HttpRequestException($"Response message: {body}"));
+
+                    throw new ReportPortalException(
+                        response.StatusCode, 
+                        response.RequestMessage.RequestUri,
+                        response.RequestMessage.Method,
+                        new HttpRequestException($"Response message: {body}"));
```

Comment:
Probably it would be good to expose it as a property rather than inner exception. It will make stacktrace shorter.

Reply:
Agree with you. fixed.